submissions/Lu-sure/frogger/app.js

```diff
@@ -0,0 +1,133 @@
+const BLOCK_WIDTH = 101;
+const BLOCK_HEIGHT = 83;
+
+const FIELD_WIDTH = 5 * BLOCK_WIDTH;
+const FIELD_HEIGHT = 6 * BLOCK_HEIGHT;
+
+const HEIGHT_SHIFT = 18;
+
+const MOVE_AREA = {width: {start: 0 , end: FIELD_WIDTH - BLOCK_WIDTH}, height: {start: FIELD_HEIGHT - BLOCK_HEIGHT - HEIGHT_SHIFT, end: 0}};
+
+const ENEMY_CONFIG = {
+    sprite: 'images/enemy-bug.png',
+    startPointX: - BLOCK_WIDTH,
+    linesToGo: {first: BLOCK_HEIGHT - HEIGHT_SHIFT, second: 2* BLOCK_HEIGHT - HEIGHT_SHIFT, third: 3* BLOCK_HEIGHT - HEIGHT_SHIFT},
+    minSpeed: 100,
+    maxSpeed: 450
+};
+
+const PLAYER_CONFIG = {
+    sprite: 'images/char-boy.png',
+    startPos: {x: 2 * BLOCK_WIDTH, y: MOVE_AREA.height.start},
+    winPositionY: BLOCK_HEIGHT-HEIGHT_SHIFT,
+    thickness: 60
+};
+
+const DELAY_OF_COLLISION = 50;
+const DELAY_OF_WIN = 200;
+
+function getRandomInteger(min, max) {
+    return Math.floor(Math.random() * (max - min + 1)) + min;
+  }
+
+const Hero = function(sprite) {
+    this.sprite = sprite;
+};
+
+Hero.prototype.render = function () {
+    ctx.drawImage(Resources.get(this.sprite), this.x, this.y);
+};
+
+const Enemy = function (y, speed, sprite) {
+    Hero.call(this, sprite);
+    this.x = ENEMY_CONFIG.startPointX;
+    this.y = y;
+    this.speed = speed;
+};
+
+Enemy.prototype = Object.create(Hero.prototype);
+Enemy.prototype.constructor = Enemy;
+
+Enemy.prototype.update = function (dt) {
+    this.x += this.speed * dt;
+    if (this.x > MOVE_AREA.width.end)
+        this.x = ENEMY_CONFIG.startPointX;
+    checkCollision();
```

Comment:
Not a really good approach, because at every enemy's every move `checkCollision` compares all enemies against the player. So number of comparisons is `numberOfEnemies ** 2`. This will pose a performance issue if number of enemies grows.

A better approach is when every enemy checks itself only against player for collision.

Reply:
Fixed, review, please)